src/components/LayerUI.tsx

```diff
@@ -111,8 +111,8 @@ const LayerUI = ({
 }: LayerUIProps) => {
   const device = useDevice();
 
-  const childrenComponents =
-    ReactChildrenToObject<UIChildrenComponents>(children);
+  const [childrenComponents, restChildren] =
+    partitionKnownChildren<UIChildrenComponents>(children);
```

Comment:
not the best API and naming (used from lodash), but didn't come up with a better one